### ISSUE TITLE
Support dots in secret key's for Secret classes

### ISSUE THREAD

**Issue by MisterSpicy** [spring-projects/spring-vault#210]
I'm trying to use the Secret auto configuration to populate a class to read configurations from Vault.  Unfortunately I don't have control over the naming conventions of the keys or paths in Vault, so many of the keys contain dots - see example

```
Key             	Value
---             	-----
jdbc.database   	test_db
jdbc.password   	ABC123
jdbc.url        	jdbc:mysql://127.0.0.1:3306/test_db
jdbc.username   	default
```

It looks like this is similar to how you cannot use a YAML configuration with dots in the keys in Spring Boot(without treating them as strings).  Is this the case with the Vault configuration as well?

If I were to put a feature request for it, it would be an Annotation to name the vault key, rather than relying variable names to match.

Thanks!

**Comment by mp911de** [comment]
What have you tried so far? Property injection via `@Value("${encrypted.value}")` works fine with Spring Vault and a regular Spring application.

**Comment by mp911de** [comment]
Closing due to lack of requested feedback. If you would like us to look at this issue, please provide the requested information and we will re-open the issue.

**Comment by MisterSpicy** [comment]
Apologies for the ultra delayed responses - I believe I have figured out where my confusion was.  I was combining the two separate methods of dealing with Vault secrets that are generally provided by Spring Vault - using a VaultTemplate/VaultOperations, and using the VaultPropertySource method.  Maybe it is possible to combine the two elegantly, but I haven't gotten to that point.

What I was attempting to do is use the VaultTemplate feature to map those values into a POJO somewhat automatically, thinking I could do that with the `@Secret` annotation and some configuration voodoo.  It's probably not voodoo, but I'm far from an expert in Spring Boot.

I was able to successfully populate the values using ValueConfiguration with the `@Value("${jdbc.username}")` annotations when setting up Vault as a  `@VaultPropertySource`.  I was also able to separately use the `VaultTemplate` and `VaultOperations` methods with a POJO to read the same configurations when annotating with Jackson annotations `@JsonProperty("jdbc.username") String dbUsername;`

The logical leap was realizing that Spring Vault communicated with the server with Jackson/JSON as noted in the documentation in one place here: https://docs.spring.io/spring-vault/docs/current/reference/html/index.html#vault.repositories.mapping
